ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -619,6 +669,31 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
       return researchPurpose.researchOutcomeList && researchPurpose.researchOutcomeList.length !== 0 ;
     }
 
+
+    updateResearchPurposeSummary(purposeSummary, value) {
+      if (this.state.researchSummaryProps[purposeSummary].warningMsg && value.length >= 50 ) {
+        this.setState(fp.set(['researchSummaryProps', purposeSummary, 'warningMsg'], false));
+      }
+      if (value.length > 1000) {
+        value = value.substring(0, 1000);
+      }
+      if (value.length >= 950 && this.state.researchSummaryProps[purposeSummary].textColor !== colors.danger) {
+        this.setState(fp.set(['researchSummaryProps', purposeSummary, 'textColor'], colors.danger));
+      } else if (value.length < 950 && this.state.researchSummaryProps[purposeSummary].textColor !== colors.disabled) {
+        this.setState(fp.set(['researchSummaryProps', purposeSummary, 'textColor'], colors.disabled));
```

Comment:
This won't make the message go away, right? It'll just turn it grey.

Reply:
ok so (i'll update the description) if the user is down to its last 50 characters then right side text characters remaining will turn red